wasm2c/wasm-rt-impl.c

```diff
@@ -202,6 +221,29 @@ static void os_install_signal_handler(void) {
   }
 }
 
+static void os_disable_and_deallocate_altstack(void) {
+  /* verify altstack was still in place */
+  stack_t ss;
+  if (sigaltstack(NULL, &ss) != 0) {
+    perror("sigaltstack failed");
+    abort();
+  }
+  if (!(ss.ss_flags & SS_DISABLE) && (ss.ss_size == SIGSTKSZ)) {
+    /* disable and free */
+    void* alt_stack = ss.ss_sp;
+    ss.ss_flags = SS_DISABLE;
+    if (sigaltstack(&ss, NULL) != 0) {
+      perror("sigaltstack failed");
+      abort();
+    }
+    assert(!os_has_altstack_installed());
+    free(alt_stack);
+    return;
+  }
+
+  fprintf(stderr, "Warning: wasm2c altstack was modified unexpectedly\n");
```

Comment:
Can you invert the check above and put this error in an early return.

Also, do we want this error string in release builds?

Reply:
Done.